lighthouse-core/audits/uses-rel-preload.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * @license Copyright 2017 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+
+'use strict';
+
+const Audit = require('./audit');
+const Util = require('../report/v2/renderer/util');
+const UnusedBytes = require('./byte-efficiency/byte-efficiency-audit');
+
+class UsesRelPreloadAudit extends Audit {
+  /**
+   * @return {!AuditMeta}
+   */
+  static get meta() {
+    return {
+      category: 'Performance',
+      name: 'uses-rel-preload',
+      description: 'Preload key requests',
+      helpText: 'Consider using <link rel=preload> to prioritize fetching late-discovered ' +
+        'resources sooner [Learn more](https://developers.google.com/web/updates/2016/03/link-rel-preload).',
+      requiredArtifacts: ['devtoolsLogs'],
+      scoringMode: Audit.SCORING_MODES.NUMERIC,
+    };
+  }
+
+  static _flattenRequests(chains, maxLevel, minLevel = 0) {
+    const requests = [];
+    const flatten = (chains, level) => {
+      Object.keys(chains).forEach(chain => {
+        if (chains[chain]) {
+          const currentChain = chains[chain];
+          if (level >= minLevel) {
+            requests.push(currentChain.request);
+          }
+
+          if (level < maxLevel) {
+            flatten(currentChain.children, level + 1);
+          }
+        }
+      });
+    };
+
+    flatten(chains, 0);
+
+    return requests;
+  }
+
+  /**
+   * @param {!Artifacts} artifacts
+   * @return {!AuditResult}
+   */
+  static audit(artifacts) {
+    const devtoolsLogs = artifacts.devtoolsLogs[UsesRelPreloadAudit.DEFAULT_PASS];
+
+    return Promise.all([
+      artifacts.requestCriticalRequestChains(devtoolsLogs),
+      artifacts.requestMainResource(devtoolsLogs),
+    ]).then(([critChains, mainResource]) => {
+      const results = [];
+      let totalWastedMs = 0;
+      // get all critical requests 2 levels deep
+      const criticalRequests = UsesRelPreloadAudit._flattenRequests(critChains, 2, 2);
+      criticalRequests.forEach(request => {
+        const networkRecord = request;
+        if (!networkRecord._isLinkPreload && networkRecord.protocol !== 'data') {
+          // calculate time between mainresource.endTime and resource start time
+          const wastedMs = (request._startTime - mainResource._endTime) * 1000;
+          totalWastedMs += wastedMs;
```

Comment:
the summing of this wastedMs is catching up to us. CNN doesn't really have 37s of opportunity here. 

![image](https://user-images.githubusercontent.com/39191/34634560-0d071b4c-f23a-11e7-96bd-e76b5ea4392a.png)

i'm fine with a number of diff solutions to this, but we should benchmark it against http://cnn.com (which probably also hits that redirect bug). :)

Reply:
yeah we should do what we do for the render-blocking audit, use maxVal - minVal for top savings

Reply:
@patrickhulce had a suggestion on what to do here. i think its the same approach as renderblockingscripts?